Civi/Share/Change.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Civi\Share;
+
+use Civi\Api4\ShareChange;
+
+class Change {
+
+  public const STATUS_LOCAL = 'LOCAL';
+
+  protected ?int $id = NULL;
+
+  protected string $type;
+
+  protected \DateTime $changedDate;
+
+  protected \DateTime $receivedDate;
+
+  protected string $entity;
+
+  protected int $entityId;
+
+  protected array $attributeChanges;
+
+  protected int $sourceNodeId;
+
+  public function __construct(
+    string $type,
+    string $entity,
+    string $entityId,
+    int $sourceNodeId,
+    array $attributeChanges = [],
+    ?\DateTime $changedDate = NULL,
+    ?\DateTime $receivedDate = NULL,
+    ?int $id = NULL
+  ) {
+    $this->type = $type;
+    $this->entity = $entity;
+    $this->entityId = $entityId;
+    $this->sourceNodeId = $sourceNodeId;
+    $this->attributeChanges = $attributeChanges;
+    $this->changedDate = $changedDate ?? new \DateTime();
+    $this->receivedDate = $receivedDate ?? new \DateTime();
+    $this->id = $id;
+  }
+
+  public static function createFromSerialized(array $serializedChange, int $sourceNodeId, ?\DateTime $receivedDate = NULL): self {
+    if (array_diff_key(
```

Comment:
Instead of `array_diff_key()` and `array_flip()` I would find `array_diff()` with `array_keys($serializedChange)` more readable.